src/Shipping/ShippingZone.php

```diff
@@ -101,10 +183,14 @@ protected function parse_shipping_zones(): void {
 			}
 
 			foreach ( $methods as $method ) {
-				// Check if the method is supported and return its properties.
+				// Skip if the method is unsupported or disabled.
+				if ( ! $method->is_enabled() || ! self::is_shipping_method_supported( $method->id ) ) {
```

Comment:
I think it's okay to leave the filtering in there because it's parsing the shipping zones and not the methods. It's actually already filtering those that it can't support. The main task here is to go through the WooCommerce shipping zone settings and return the methods that we support in the format that we expect.